### PR TITLE
dotnet CLI preview 2 doesn't have the quiet option anymore

### PR DESCRIPTION
**Pull request by adamhathcock**
Just noticed this.

### CODE REVIEW

**Comment by adamhathcock** [comment]
Looks like the builds are failing because your build script on the dotnet cli also uses the Quiet option.

**Comment by patriksvensson** [comment]
@adamhathcock Great catching this!

The reason for the builds failing is that the tests for DotNetRestore has not been updated.
I'm not sure if we want to remove the property directly due to backwards compatibility. I think we would want to obsolete it with a warning for now and refer to using the Verbosity parameter and remove it in a couple releases.

**Comment by adamhathcock** [comment]
I'll update

**Comment by adamhathcock** [comment]
Now it's failing because of Warning As Error.  Should I just remove the Quiet attribute?

**Comment by patriksvensson** [comment]
@adamhathcock Yes. I think you can remove that attribute for now and just write to the log in the Tool (as a warning) that the flag has been removed from dotnet cli RTM and they should set verbosity instead.

**Comment by devlead** [comment]
Isn't this just that the tests should pragma ignore the obsolete attribute?

**Comment by patriksvensson** [comment]
@devlead @adamhathcock Yes, that would be preferred.

File: src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestoreSettings.cs [hunk 1]
```diff
@@ -38,7 +40,8 @@ public sealed class DotNetCoreRestoreSettings : DotNetCoreSettings
 
         /// <summary>
         /// Gets or sets a value indicating whether to display any output.
-        /// </summary>
+        /// </summary>
+        [Obsolete("dotnet cli does not have this option")]
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestoreSettings.cs]: Can we change this to `.NET CLI does not support this option anymore. Please use DotNetCoreRestoreSettings.Verbosity instead.`

**Comment by adamhathcock** [comment]
finally did the pragmas in all the correct places.

File: src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestorer.cs [hunk 2]
```diff
@@ -101,12 +101,13 @@ private ProcessArgumentBuilder GetArguments(string root, DotNetCoreRestoreSettin
...
                 }
             }
-
+#pragma warning disable 0618
             // Quiet
-            if (settings.Quiet)
+            if (settings.Quiet && !settings.Verbosity.HasValue)
             {
-                builder.Append("--quiet");
+                settings.Verbosity = DotNetCoreRestoreVerbosity.Minimal;
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestorer.cs]: I don't think we should set this propery automatically. Better to write to the log as a warning giving the same hint as the obsolete message.

**Comment by gep13** [comment]
@adamhathcock are you in a position to update with the comment that was made by @patriksvensson

**Comment by adamhathcock** [comment]
I've been on vacation for the past two weeks.  I can fix it on Wednesday

**Comment by adamhathcock** [comment]
Looking at this now: I need a `ICakeContext` to do a log item but the infrastructure doesn't look well supported to that from a tool.

Making the tool take the context involves changing the tests a lot at least.

**Comment by adamhathcock** [comment]
Nevermind, I was being dumb.  I think I've done everything requested....but now I just noticed I have to update the branch again

**Comment by adamhathcock** [comment]
@gep13 @patriksvensson finally updated and passed.  Needs a small review again

**Comment by adamhathcock** [comment]
@gep13 should I create an issue?

**Comment by gep13** [comment]
@adamhathcock yes, if you could.  Ideally, an issue is created, which gives us a place to discuss the problem/feature before any work is undertaken, and then we use the issue title to automatically generate the release notes for each release.

**Comment by adamhathcock** [comment]
I thought it was a simple fix so I didn't...next time I promise I will :)

**Comment by gep13** [comment]
>@adamhathcock said...
I thought it was a simple fix so I didn't...next time I promise I will :)

Not a problem at all.  For anything that is a documentation change, we typically don't want/need issues created, however, anything that is a code change, we would like to see an issue getting raised/discussed first, just in case something is already being worked on, or not required, etc.  The last thing we want is people doing work that isn't required, and then having to close the PR.  Obviously, that won't be the case here, but we like to follow the same process, so that there is some consistency, and people know what to expect.

**Comment by gep13** [comment]
Relates to #1073

**Comment by gep13** [comment]
LGTM! Thanks again for the contribution.

**Pull request merged**